api/v1alpha3/eraserconfig_types.go

```diff
@@ -0,0 +1,209 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha3
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type (
+	Duration       time.Duration
+	Runtime        string
+	RuntimeAddress string
+
+	RuntimeSpec struct {
+		Name    Runtime        `json:"name"`
+		Address RuntimeAddress `json:"address"`
+	}
+)
+
+const (
+	RuntimeContainerd Runtime = "containerd"
+	RuntimeDockerShim Runtime = "dockershim"
+	RuntimeCrio       Runtime = "crio"
+
+	ContainerdPath = "/run/containerd/containerd.sock"
+	DockerPath     = "/run/dockershim.sock"
+	CrioPath       = "/run/crio/crio.sock"
+)
+
+func ConvertRuntimeToRuntimeSpec(r Runtime) (RuntimeSpec, error) {
+	var rs RuntimeSpec
+
+	switch rt := Runtime(r); rt {
```

Comment:
The cast to the `Runtime` type is unnecessary because `r` is already a `Runtime (it was wrong in my draft PR too).